thcrap_tsa/src/ascii.cpp

```diff
@@ -255,6 +255,54 @@ int ascii_vpatchf_th06(
 #undef SCORE_LEN
 }
 
+int ascii_vpatchf_th07_th08(
+	ascii_put_func_t& putfunc, vector3_t pos, const char* fmt, va_list va
+)
+{
+	auto id = strings_id(fmt);
+	auto single_str = strings_vsprintf((size_t) &pos, fmt, va);
+	if (!id) {
+		return putfunc(params.ClassPtr(), pos, single_str);
+	}
+
+	// Since ZUN didn't bother centering these strings, a correction must be
+	// applied to each one in addition to the usual center alignment.
+	// Since the spell card bonus doesn't move, I just centered it outright.
+	if (!strcmp(id, "th07 Full Power")) {
```

Comment:
I think all this if/else if chain could be replaced with an array and a loop. Something like:
```c
struct string_to_center
{
  const char *id;
  const char *str;
  float correction;
};

string_to_center var_name[] = {
  { "th07 Full Power", "Full Power Mode!", game_id == TH07 ? 8.5f : 15.5f },
  { "th06_ascii_bonus_format", "BONUS 12345678", 0 },
  // ...
};
```

th06_ascii_centered_spell_bonus is different, but all the other strings would fit in this format.

Also, all these English strings, they're translatable, aren't they? Wouldn't this centering code fail if they're translated?

Reply:
(Accidentally resolved) I personally don't think the string solution is that much more elegant, but if you want to use that, feel free to do so.

The centering has to take the original string into account to align the new string properly. I've tested it with my croatian translation and it works fine: [before](https://media.discordapp.net/attachments/253873278689607680/1146792309942390985/image.png) [after](https://media.discordapp.net/attachments/1091326716829511750/1149481414237110332/2023-09-08_01_03_17-Touhou__Sablasni_san__Savrsen_tresnjin_cvijet_v1.00b.png)

Reply:
Okay for the English strings.

Waiting for other people's opinion on `if` vs array. IMO the array version makes more clear that almost all of them share the same code.

Reply:
Changed my mind, I pushed the array version now.